### DISCUSSION TITLE
Page-level events

### DISCUSSION THREAD

**Discussion by me21** [zauberzeug/nicegui#3134] [Ideas / Feature Requests]
The documentation lists application-level events such as on_startup, on_connect, on_disconnect. Would it be possible to introduce page-level events?

For example, I would like to subscribe to MQTT  messages only if someone is viewing a specific page of my app and to unsubscribe otherwise. I could maintain a list of viewers with on_connect/disconnect events triggered by that page only if they were available.

Maybe this can be already achieved somehow?

**Comment by falkoschindler** [comment]
Hi @me21,

A "page" is usually represented by a `client` object, which already provides connect and disconnect events:
```py
ui.context.client.on_connect(lambda: print('Connected!'))
ui.context.client.on_disconnect(lambda: print('Disconnected!'))
```
